hashintel/hash
# Add HashiCorp Vault to HASH developer environment

## 🌟 What is the purpose of this PR?




This PR adds HashiCorp Vault to the developer external services of the HASH app, and adds a basic vault client to the api (which only initializes if the required environment variables are provided)

## 🔗 Related links





- ...

## 🔍 What does this change?




See commits

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:



### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:



## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



None

## ❓ How to test this?



1. Add the following to `apps/hash-api/src/index.ts` after the vault client initialization:
```TypeScript
  if (vaultClient) {
    const createdSecret = await vaultClient.write<{ value: string }>({
      secretMountPath: "secret",
      path: "example",
      data: { value: "hello world" },
    });

    console.log({ createdSecret });

    const retrievedValue = await vaultClient.read({
      secretMountPath: "secret",
      path: "example",
    });

    console.log({ retrievedValue });
  }
```
2. Run the external services and HASH, and observe the output in the console

## 📹 Demo



<img width="529" alt="image" src="https://github.com/hashintel/hash/assets/42802102/679ece6e-1b2b-4724-b466-de2e1593a956">
